bazel/repositories.bzl

```diff
@@ -315,13 +316,14 @@ def _com_github_libevent_libevent():
 
 def _net_zlib():
     location = REPOSITORY_LOCATIONS["net_zlib"]
-    http_archive(
+
+    # Use the dev branch of zlib to resolve fuzz bugs and out of bound
+    # errors resulting in crashes in zlib 1.2.11.
+    # TODO(asraa): Remove when zlib > 1.2.11 is released.
+    new_git_repository(
         name = "net_zlib",
+        branch = "develop",
```

Comment:
Can you follow the pattern for other dependencies such as `com_google_googletest`, where we use a GitHub archival URL and `http_archive` as before? We tend to pin to a specific git commit and hash, rather than branch, to ensure we have stable and reproducible builds, with SHA256 hashes to protect.

Reply:
@asraa the standard pattern for the latest dev branch commit of zlib would be:
```
    net_zlib = dict(
        sha256 = "155a8f8c1a753fb05b16a1b0cc0a0a9f61a78e245f9e0da483d13043b3bcbf2e",
        strip_prefix = "zlib-79baebe50e4d6b73ae1f8b603f0ef41300110aa3",
        # 2019-04-14 development branch
        urls = ["https://github.com/madler/zlib/archive/79baebe50e4d6b73ae1f8b603f0ef41300110aa3.tar.gz"],
    ),
```
With the following diff this compiles on master
```diff
diff --git a/bazel/repositories.bzl b/bazel/repositories.bzl
index d02e30bf7..87d008ef2 100644
--- a/bazel/repositories.bzl
+++ b/bazel/repositories.bzl
@@ -318,10 +318,6 @@ def _net_zlib():
     http_archive(
         name = "net_zlib",
         build_file_content = BUILD_ALL_CONTENT,
-        # The patch is only needed due to https://github.com/madler/zlib/pull/420
-        # TODO(htuch): remove this when zlib #420 merges.
-        patch_args = ["-p1"],
-        patches = ["@envoy//bazel/foreign_cc:zlib.patch"],
         **location
     )
     native.bind(
```

Reply:
Changes LGTM with the latest commit :+1: